### DISCUSSION TITLE
Aggregating custom fields

### DISCUSSION THREAD

**Discussion by GP4cK** [TriPSs/nestjs-query#185] [Q&A] [answered]
Hello!
The below example is based on [the doc about aggregation](https://tripss.github.io/nestjs-query/docs/graphql/aggregations).

Lets say I wanted to let users define custom fields for all TodoItems (ex: `cost` and `dueDate`). I'm thinking I could:
1. Create a `CustomFields` table with id, type, rank etc.
2. Add a JSON column to the `TodoItem` entity so a TodoItem would look like this:

```ts
{
  id: 1,
  title: 'Fix the washing machine',
  // other fields...
  custom: {
    cost: 100,
    dueDate: 25/09/2023
  }
}
```

When I enable the aggregation, it would be really cool if it could generate some kind of custom field so that I could for example, compute the total cost of the TodoItem.

```diff
type TodoItemSumAggregate {
  id: Float
  priority: Float
+++  custom(customFieldID: String!): Float
}
```

Then, the graphql query would look like:

```graphql
{
  todoItemAggregate {
    sum {
      custom(customFieldID: "cost")
    }
}
```

And then somewhere, I would create my own `AggregateBuilder` to resolve the `custom(customFieldID: "cost")` and add a JSON selector in the postgres query.

Has anyone faced a similar use case?

**Comment by TriPSs** [comment, marked as answer]
You could add the field with an `@FilterableField` to the DTO definition used by the aggregation and than overwrite the resolver / service to fetch the data the correct way.
